packages/fuels-core/src/types/transaction_builders.rs

```diff
@@ -193,9 +193,15 @@ macro_rules! impl_tx_trait {
 
         impl $ty {
             fn is_using_predicates(&self) -> bool {
-                self.inputs()
-                    .iter()
-                    .any(|input| matches!(input, Input::ResourcePredicate { .. }))
+                self.inputs().iter().any(|input| {
+                    matches!(
+                        input,
+                        Input::ResourcePredicate { .. }
+                            | Input::FuelInput(FuelInput::CoinPredicate(_))
```

Comment:
We should not be mixing the SDK `Input` with the `FuelInput` there is a reason why we made this separation